crates/sui-indexer-alt/src/handlers/sum_obj_types.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::{btree_map::Entry, BTreeMap},
+    sync::Arc,
+};
+
+use anyhow::{anyhow, ensure};
+use diesel::{upsert::excluded, ExpressionMethods};
+use diesel_async::RunQueryDsl;
+use futures::future::try_join_all;
+use sui_types::{
+    effects::TransactionEffectsAPI, full_checkpoint_content::CheckpointData, object::Owner,
+};
+
+use crate::{
+    db,
+    models::objects::{StoredOwnerKind, StoredSumObjType},
+    pipeline::{sequential::Handler, Processor},
+    schema::sum_obj_types,
+};
+
+/// Each insert or update will include at most this many rows -- the size is chosen to maximize the
+/// rows without hitting the limit on bind parameters.
+const UPDATE_CHUNK_ROWS: usize = i16::MAX as usize / 8;
```

Comment:
would be good to read the number of columns 8 from the schema on the starting phase of the handler, in case of schema changes especially adding new column?

Reply:
Addressed in #20070